query/src/pipelines/new/processors/transforms/hash_join/row.rs

```diff
@@ -155,8 +152,7 @@ pub fn compare_and_combine(
 
     let data_block = DataBlock::create(Arc::new(DataSchema::new(data_fields)), columns);
 
-    let filter =
-        ExpressionEvaluator::eval(ctx.try_get_function_context()?, &predicate, &data_block)?;
+    let filter = ExpressionEvaluator::eval(func_ctx, &predicate, &data_block)?;
```

Comment:
There is no need to filter the result anymore, since the hash keys should have been compared when probing the hash table.

Reply:
Nice!!! I need to update the performance result 😆